Switched back to normal Lensfun search mode for lens lookups.

bronger opened the pull request:
The option LF_SEARCH_SORT_AND_UNIQUIFY was needed when darktable checked for
uniqueness of the search result (see bug #9662).  However, since commit
362ba6a, these checks have been removed, and now, LF_SEARCH_SORT_AND_UNIQUIFY
is harmful in case of lookups.  The reason is that in contrast to what commit
message 362ba6a says, the exact lens model name may not be the very first
returned element.

LebedevRI commented:
@bronger 
Could you please add a bit more context?
Does this fix some issue with lens lookup?
With default lens autodetection?

Is this something we should want in 2.0, or it should wait for 2.1?

bronger commented:
The option LF_SEARCH_SORT_AND_UNIQUIFY sorts by focal length, and it collapses entries with the same lens model name, keeping the one that has the higher matching score.  Instead, if you pass "0", all matching lenses are returned sorted by decreasing matching score.  The first mode is best if you want to generate a nice-look selection list, the latter mode is best if you want to lookup a particular lens.

Then why do we use LF_SEARCH_SORT_AND_UNIQUIFY everywhere currently?  Because formely, DT insisted on uniqueness of the search result when doing a lens lookup.  This was bad because Lensfun quite often finds more than one matching lens, e.g. different crops of the same lens.  So I added LF_SEARCH_SORT_AND_UNIQUIFY.  After that, I found out that Lensfun may *still* return more than one entry, namely if there is a lens model name slightly longer than the requested one, e.g. "Olympus Zuiko ..." and "Olympus M.Zuiko".  So, I removed the uniqueness checks completely.

Now, we have a problem in Git Lensfun with images taken with "Olympus Zuiko Digital ED 9-18mm f/4.0-5.6".  DT insists on using "Olympus M.Zuiko Digital ED 9-18mm f/4.0-5.6", even if you manually select the first lens!  This is because "M" is before "Z" in the alphabet, and LF_SEARCH_SORT_AND_UNIQUIFY sorts by focal length, then lens name.

Thus, this PR changes back to ordinary "sort by matching score" in places where DT just wants to lookup a lens, and keeps LF_SEARCH_SORT_AND_UNIQUIFY where DT builds actually a list.

LebedevRI commented:
Okay I hope this does not cause any regressions :)
